src/docfx/template/TemplateEngine.cs

```diff
@@ -195,12 +196,18 @@ public string ProcessHtml(ErrorBuilder errors, FilePath file, string html)
         return LocalizationUtility.AddLeftToRightMarker(_cultureInfo, result);
     }
 
-    private string CreateContent(FilePath file, string? mime, JObject pageModel)
+    private string CreateContent(FilePath file, string? mime, ref JObject pageModel)
```

Comment:
This method shouldn't be mutating the `pageModel`.
```suggestion
    private string CreateContent(FilePath file, string? mime, JObject pageModel)
```

Reply:
right